packages/hop-node/src/watchers/BondTransferRootWatcher.ts

```diff
@@ -232,6 +257,85 @@ class BondTransferRootWatcher extends BaseWatcher {
+    await this.validateDestinationChainId(txParams)
+    await this.validateUniqueness(txParams)
+  }
+
+  async validateDbExistence (txParams: SendBondTransferRootTxParams): Promise<void> {
+    // Validate DB existence with calculated transferRootId
+    const calculatedDbTransferRoot = await this.getCalculatedDbTransferRoot(txParams)
+    if (calculatedDbTransferRoot?.transferRootId !== txParams.transferRootId) {
+      throw new PreTransactionValidationError(`Calculated calculatedTransferRootId (${calculatedDbTransferRoot?.transferRootId}) does not match transferRootId in db`)
+    }
+  }
+
+  async validateDestinationChainId (txParams: SendBondTransferRootTxParams): Promise<void> {
+    // Validate that the destination chain id matches the db entry
+    const calculatedDbTransferRoot = await this.getCalculatedDbTransferRoot(txParams)
+    if (calculatedDbTransferRoot?.destinationChainId !== txParams.destinationChainId) {
+      throw new PreTransactionValidationError(`Calculated destinationChainId (${txParams.destinationChainId}) does not match destinationChainId in db (${calculatedDbTransferRoot?.destinationChainId})`)
+    }
+  }
+
+  async validateUniqueness (txParams: SendBondTransferRootTxParams): Promise<void> {
+    // Validate uniqueness for redundant reorg protection. A transferId should only exist in one transferRoot per source chain
+    const transferIds = txParams.transferIds.map((x: string) => x.toLowerCase())
+
+    const dbTransferRoots: TransferRoot[] = (await this.db.transferRoots.getTransferRootsFromTwoWeeks())
+      .filter(dbTransferRoot => dbTransferRoot.transferRootId !== txParams.transferRootId)
+      .filter(dbTransferRoot => dbTransferRoot.sourceChainId === this.bridge.chainId)
+    const dbTransferIds: string[] = dbTransferRoots.flatMap(dbTransferRoot => dbTransferRoot.transferIds!)
+
+    for (const transferId of transferIds) {
+      const transferIdCount: string[] = dbTransferIds.filter((dbTransferId: string) => dbTransferId.toLowerCase() === transferId)
+      if (transferIdCount.length > 0) {
+        throw new PreTransactionValidationError(`transferId (${transferId}) exists in multiple transferRoots in db`)
+      }
+    }
+  }
+
+  async validateLogsWithBackupRpc (txParams: SendBondTransferRootTxParams): Promise<void> {
+    // Validate logs with backup RPC endpoint, if it exists
+    const calculatedDbTransferRoot = await this.getCalculatedDbTransferRoot(txParams)
+    const blockNumber = calculatedDbTransferRoot.commitTxBlockNumber
+
+    const redundantRpcUrls = getRedundantRpcUrls(this.chainSlug) || []
+    for (const redundantRpcUrl of redundantRpcUrls) {
+      const redundantProvider = new providers.JsonRpcProvider(redundantRpcUrl)
+
+      // TODO: Better way to do this
+      const l2Bridge = contracts.get(this.tokenSymbol, this.chainSlug)?.l2Bridge
+      const events = await l2Bridge.connect(redundantProvider).queryFilter(
+        l2Bridge.filters.TransfersCommitted(),
```

Comment:
would it make sense to get event using `rootHash` and `destinationChainId` in filter?

```suggestion
        l2Bridge.filters.TransfersCommitted(txParams.transferRootHash, txParams.destinationChainId),
```